text/0000-informational-commands.md

```diff
@@ -0,0 +1,775 @@
+runtime node@v12.2.0
+runtime node@v11.9.0
+runtime node@v10.15.3 (default)
+runtime node@v8.16.0
+packager yarn@v1.16.0 (default)
+packager yarn@v1.12.3
+tool create-react-app / create-react-app@v3.0.1 node@v12.2.0 npm@built-in (default)
+tool ember / ember-cli@v3.10.0 node@v12.2.0 npm@built-in (default)
+tool ember / ember-cli@v3.8.2 node@v12.2.0 npm@built-in
+tool tsc / typescript@v3.4.5 node@v12.2.0 npm@built-in
+tool tsserver / typescript@v3.4.5 node@v12.2.0 npm@built-in
+tool tsc / typescript@v3.0.3 node@v12.2.0 npm@built-in (default)
+tool tsserver / typescript@v3.0.3 node@v12.2.0 npm@built-in (default)
+tool yarn-deduplicate / yarn-deduplicate@v1.1.1 node@v12.2.0 npm@built-in
+```
+
+</details>
+
+<details><summary>In the <code>node-only</code> project</summary>
+
+```sh
+$ volta list all --plain
+runtime node@v12.2.0
+runtime node@v11.9.0
+runtime node@v10.15.3 (default)
+runtime node@v8.16.0 (current @ ~/node-only/package.json)
+packager yarn@v1.16.0 (default)
+packager yarn@v1.12.3
+tool create-react-app / create-react-app@v3.0.1 node@v12.2.0 npm@built-in (default)
+tool ember / ember-cli@v3.10.0 node@v12.2.0 npm@built-in (default)
+tool ember / ember-cli@v3.8.2 node@v12.2.0 npm@built-in
+tool tsc / typescript@v3.4.5 node@v12.2.0 npm@built-in
+tool tsserver / typescript@v3.4.5 node@v12.2.0 npm@built-in
+tool tsc / typescript@v3.0.3 node@v12.2.0 npm@built-in (default)
+tool tsserver / typescript@v3.0.3 node@v12.2.0 npm@built-in (default)
+tool yarn-deduplicate / yarn-deduplicate@v1.1.1 node@v12.2.0 npm@built-in
+```
+
+</details>
+
+<details><summary>In the <code>node-and-yarn</code> project</summary>
+
+```sh
+$ volta list all --plain
+runtime node@v12.2.0 (current @ ~/node-and-yarn/project.json)
+runtime node@v11.9.0
+runtime node@v10.15.3 (default)
+runtime node@v8.16.0
+packager yarn@v1.16.0 (default)
+packager yarn@v1.12.3 (current @ ~/node-and-yarn/project.json)
+tool create-react-app / create-react-app@v3.0.1 node@v12.2.0 npm@built-in (default)
+tool ember / ember-cli@v3.10.0 node@v12.2.0 npm@built-in (default)
+tool ember / ember-cli@v3.8.2 node@v12.2.0 npm@built-in
+tool tsc / typescript@v3.4.5 node@v12.2.0 npm@built-in
+tool tsserver / typescript@v3.4.5 node@v12.2.0 npm@built-in
+tool tsc / typescript@v3.0.3 node@v12.2.0 npm@built-in (default)
+tool tsserver / typescript@v3.0.3 node@v12.2.0 npm@built-in (default)
+tool yarn-deduplicate / yarn-deduplicate@v1.1.1 node@v12.2.0 npm@built-in
+```
+
+</details>
+
+### `volta list <package>`
+
+List all fetched versions of a specific package, along with its associated binaries.
+
+#### Human
+
+The basic format is:
+
+```sh
+volta list <package> --human
+
+    <version> [(default|current @ <project path>)]
+        binaries: [<binary name>]...
+        platform:
+            runtime: node@<version>
+            packager: built-in npm|<npm|yarn>@<version>
+```
+
+For the TypeScript config specified in the canonical example:
+
+```sh
+volta list package typescript --human
```

Comment:
Slight discrepancy here between `volta list <package>` and `volta list package typescript`.

This will probably need to be `volta list package <package>` and `volta list tool <tool>` to be able to disambiguate when packages and tools have the same names (like the `create-react-app` example).

Reply:
This is actually the reason why I had `volta list package <package>` and `volta list tool <tool>` in an earlier draft of the RFC. I'm back-and-forth on it, because in the case you identified, it's the same output either way. In a case where there's more than one binary, but one does match the package name, I don't think it (harms* anything to default to the package in that case? 🤔 But I'm *very* willing to be persuaded here—I just found myself leaning *slightly* in the direction of brevity!

Reply:
After thinking about this more, I like the `volta list <package-or-tool>` approach, where it defaults to the package. It's more compact to type, and in cases where the binary and the package are the same name, you get more information.